protos/orchestrator_service.proto

```diff
@@ -117,6 +126,7 @@ message TimerCreatedEvent {
 message TimerFiredEvent {
     google.protobuf.Timestamp fireAt = 1;
     int32 timerId = 2;
+    TraceContext parentTraceContext = 3;
```

Comment:
I'm not yet 100% sure if the parent trace context needs to be saved in `TimerFiredEvent`. It seems like we should be able to infer it from the `ExecutionStartedEvent`, but I haven't yet verified this.

Reply:
I've removed this. The `durabletask-go` engine now just uses the current trace context (which happens to be from the `ExecutionStartedEvent`).